src/audio/module_adapter/module_adapter.c

```diff
@@ -903,6 +918,74 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 	return ret;
 }
 
+static inline int module_adapter_simple_copy_one_source_one_sink(struct processing_module *mod)
+{
+	struct comp_buffer __sparse_cache *source_c;
+	struct comp_buffer __sparse_cache *sink_c;
+	struct comp_dev *dev = mod->dev;
+	uint32_t source_avail_bytes;
+	uint32_t sink_free_bytes;
+	int ret = 0;
+
+	source_c = buffer_acquire(mod->source);
+	sink_c = buffer_acquire(mod->sink);
+
+	/* TODO: Add to prepare check that mod->period_bytes is not conflicting with
+	 * client's align requirements.
+	 */
+
+	/* TODO: Add to prepare check that mod->period_bytes and dev->frames match. */
+
+	source_avail_bytes = audio_stream_get_avail_bytes(&source_c->stream);
+	sink_free_bytes = audio_stream_get_free_bytes(&sink_c->stream);
+
+	//comp_info(dev, "dev_frames=%d %d %d %d", dev->frames, mod->period_bytes,
+	//	  source_avail_bytes, sink_free_bytes);
+
+	if (source_avail_bytes < mod->period_bytes || sink_free_bytes < mod->period_bytes)
```

Comment:
We could set mod->source or mod->sink to NULL when the buffer is not existing or don't care as with mixout. Would that make sense?

Reply:
Setting here frames to zero solves the issue for mixout component. It processes the data (or actually mixin) without checking the frames count.